Editor/CoreAPI/GameLiftComputeManager.cs

```diff
@@ -18,8 +21,47 @@ public GameLiftComputeManager(IAmazonGameLiftWrapper wrapper)
         {
             _amazonGameLiftWrapper = wrapper;
         }
-        
-        public async Task<RegisterFleetComputeResponse> RegisterFleetCompute(string computeName, string fleetId, string fleetLocation,
+
+        public async Task<RegisterFleetComputeResponse> RegisterFleetCompute(string computeName, string fleetId,
+            string fleetLocation,
+            string ipAddress)
+        {
+            if (_amazonGameLiftWrapper != null)
+            {
+                if (string.IsNullOrWhiteSpace(computeName))
+                {
+                    return Response.Fail(new RegisterFleetComputeResponse
+                    {
+                        ErrorCode = ErrorCode.InvalidComputeName
+                    });
+                }
+                
+                if (string.IsNullOrWhiteSpace(ipAddress))
+                {
+                    return Response.Fail(new RegisterFleetComputeResponse
+                    {
+                        ErrorCode = ErrorCode.InvalidIpAddress
+                    });
+                }
+                
+                var registerFleetComputeResponse = await RegisterCompute(computeName, fleetId, fleetLocation, ipAddress);
+                if (registerFleetComputeResponse.Success)
+                {
+                    return Response.Ok(new RegisterFleetComputeResponse()
+                    {
+                        ComputeName = computeName,
+                        IpAddress = ipAddress,
+                        WebSocketUrl = registerFleetComputeResponse.WebSocketUrl
```

Comment:
Why isn't this handled by RegisterCompute? They both have the same return type, so this code looks very odd

Reply:
Cleaned up in error conditions pr